/*
apt install -y libprotobuf-dev protobuf-compiler
protoc --cpp_out=. --proto_path=. model_header.proto
 */
syntax = "proto3";

package newtranx.ai_server;
option  cc_enable_arenas = true;

message ModelHeader {
    string model_id = 10;       // 模型唯一标识，默认自动生成uuid
    string expire_date = 20;    // 到期时间，格式: 20250701
    uint32 n_concurrency = 30;  // 并发数
    uint64 start_pos = 40;      // 模型数据在文件中的起始位置, 即自定义header的长度
    uint64 end_pos = 50;        // 模型数据在文件中的结束位置, 在生成加密文件时自动计算
    string checksum = 60;       // 模型数据校验值，默认是原始模型文件的md5校验
    string original_filename = 70;   // 原始模型文件名
    bool   is_directory = 80;   // 原始模型以目录形式存在
}
